lib/utilities.h

```diff
@@ -140,6 +140,19 @@ metal_bitmap_next_clear_bit(unsigned long *bitmap, unsigned int start,
 	return bit;
 }
 
+static inline unsigned int
+metal_bitmap_loopnext_clear_bit(unsigned long *bitmap, unsigned int start,
+				unsigned int max)
+{
+	unsigned int bit;
+
+	for (bit = start;
+	     bit < max + start && !metal_bitmap_is_bit_clear(bitmap, bit % max);
+	     bit++)
+		;
+	return bit % max;
+}
+
```

Comment:
As Proposed @tnmysh in https://github.com/OpenAMP/open-amp/pull/514, we could be able to  improve `metal_bitmap_for_each_clear_bit` macro instead.

Here is a proposal ( not tested)

```c
static inline unsigned int
metal_bitmap_next_clear_bit(unsigned long *bitmap, unsigned int start,
			    unsigned int max)
{
	unsigned int bit;

	/* check overflow */
        metal_assert( max + start > max);

	for (bit = start;
	     bit < max + start && !metal_bitmap_is_bit_clear(bitmap, bit % max);
	     bit++)
		;

	return bit;
}
```